src/main/java/org/dependencytrack/tasks/MetricsUpdateTask.java

```diff
@@ -426,14 +427,33 @@ MetricCounters updateProjectMetrics(final QueryManager qm, final long oid) {
      * @return MetricCounters
      */
     MetricCounters updateComponentMetrics(final QueryManager qm, final long oid) {
+        final List<String> dedupedVulnIdentifiers = new ArrayList<>();
         final Component component = qm.getObjectById(Component.class, oid);
         LOGGER.debug("Executing metrics update for component: " + component.getUuid());
         final Date measuredAt = new Date();
 
         final MetricCounters counters = new MetricCounters();
         // Retrieve the non-suppressed vulnerabilities for the component
         for (final Vulnerability vuln: qm.getAllVulnerabilities(component)) {
-            counters.updateSeverity(vuln.getSeverity());
+            final String dedeupedVulnIdentifier = vuln.getSource() + "|" + vuln.getVulnId();
+            if (!dedupedVulnIdentifiers.contains(dedeupedVulnIdentifier)) {
+                // Do not count duplicate vulnerabilities with different identifiers in the overall score.d
+                boolean found = false;
+                for (final VulnerabilityAlias alias: vuln.getAliases()) {
+                    if ((dedupedVulnIdentifiers.contains(dedeupedVulnIdentifier)
+                            || (alias.getInternalId() != null && dedupedVulnIdentifiers.contains(Vulnerability.Source.INTERNAL.name() + "|" + alias.getInternalId()))
+                            || (alias.getCveId() != null && dedupedVulnIdentifiers.contains(Vulnerability.Source.NVD.name() + "|" + alias.getCveId()))
+                            || (alias.getSonatypeId() != null && dedupedVulnIdentifiers.contains(Vulnerability.Source.OSSINDEX.name() + "|" + alias.getSonatypeId()))
+                            || (alias.getGhsaId() != null && dedupedVulnIdentifiers.contains(Vulnerability.Source.GITHUB.name() + "|" + alias.getGhsaId()))
+                            || (alias.getVulnDbId() != null && dedupedVulnIdentifiers.contains(Vulnerability.Source.VULNDB.name() + "|" + alias.getVulnDbId())))) {
+                        found = true;
+                    }
+                }
+                if (!found) {
+                    counters.updateSeverity(vuln.getSeverity());
```

Comment:
This will be first-come-first-served WRT the severity. `getAllVulnerabilities` does not do any explicit sorting, resulting in the first vulnerability that was ingested to dictate the severity. This will result in different metrics across different DT instances.

Should this be more deterministic?

Reply:
We have to include GSD since its included in OSV. But we can leave our Snyk for the time being.